opm/simulators/linalg/bda/openclSolverBackend.cpp

```diff
@@ -212,6 +219,17 @@ void openclSolverBackend<block_size>::gpu_pbicgstab(WellContributions& wellContr
     queue->enqueueCopyBuffer(d_r, d_p, 0, 0, sizeof(double) * N, nullptr, &event);
     event.wait();
 
+    wellContribs.setReordering(toOrder, true);
+    wellContribs.getData(&h_valsC, &h_valsD, &h_valsB, &h_colsC, &h_colsB, &h_val_pointers);
+
+    queue->enqueueWriteBuffer(d_valsC, CL_TRUE, 0, sizeof(double) * num_blocks * dim_ * dim_wells, h_valsC);
+    queue->enqueueWriteBuffer(d_valsD, CL_TRUE, 0, sizeof(double) * num_std_wells * dim_wells * dim_wells, h_valsD);
+    queue->enqueueWriteBuffer(d_valsB, CL_TRUE, 0, sizeof(double) * num_blocks * dim_ * dim_wells, h_valsB);
+    queue->enqueueWriteBuffer(d_colsC, CL_TRUE, 0, sizeof(int) * num_blocks, h_colsC);
+    queue->enqueueWriteBuffer(d_colsB, CL_TRUE, 0, sizeof(int) * num_blocks, h_colsB);
+    queue->enqueueWriteBuffer(d_val_pointers, CL_TRUE, 0, sizeof(unsigned int) * (num_std_wells + 1), h_val_pointers, nullptr, &event);
```

Comment:
Please explain what we are doing here. It seems `getData` is actually performing a superfluous host-to-host copy. Coudn't we just pass the pointer copied from there (d_Cnnzs_ocl, ..., d_Bnnzs_ocl,  d_Ccols_ocl, ...) directly to `queue->enqueueWriteBuffer`?

Reply:
Yes, it is a host-to-host copy. I've implemented it like this because the `d_*_ocl` arrays are assembled by parts (through concatenations) in the `addMatrix` function. I couldn't see another way of passing data from the `WellContributions` object to the `openclSolverBackend` object. But I'm completely open to suggestions.

Reply:
The Opm::Wells are separate objects on the CPU. We want some of their data in a contiguous array on GPU. I only see two options:
- multiple concatenating CPU-to-GPU transfers
- multiple concatenating CPU-to-CPU transfers, preferably to pinned memory. One CPU-to-GPU transfer

Reply:
I think the last option is what I'm currently doing